### DISCUSSION TITLE
On: workflow_run: does not work for me

### DISCUSSION THREAD

**Discussion by sblantipodi** [community/community#25288] [Actions]
<p>Hi all,<br>
I have a workflow that is triggered on release</p>
<pre><code class="lang-auto">name: CreateRelease

on:
  release:
    types: [published]
</code></pre>
<p>and a workflow that should be triggered after the CreateRelease workflow runned</p>
<pre><code class="lang-auto">name: GitHub Linux Release

on:
  workflow_run:
    workflows: ["CreateRelease"]
    branches: [master]
    types:
      - completed
      - requested
</code></pre>
<p>second workflow is never triggered, why? thanks!</p>
<p>I want to create a release in a workflow triggered by a tag push,<br>
after the release has been created, I want to trigger two workflow that build the linux and windows build.</p>
<p>how can I do it?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/sblantipodi">@sblantipodi</a>,</p>
<p>Is the workflow “<strong>GitHub Linux Release</strong>” existing on the <strong>master</strong> or default branch?<br>
To trigger a workflow runs on the <strong>workflow_run</strong> event, the workflow file should exist on the <strong>master</strong> or default branch.</p>
<p>Here is a recent reported similar topic as reference:</p><aside class="quote quote-modified" data-post="8" data-topic="128345">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/workflow-runs-not-starting-after-previous-workflow-completes/128345/8">Workflow runs not starting after previous workflow completes</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    <a class="mention" href="/andrewconnell">@andrewconnell</a>, 
There are few points maybe can explain the problems: 


The workflow runs on workflow_run event must exist on the master or default branch. And this workflow will only run on the master or default branch, regardless of whether it is triggered by a workflow run on other branches or not. 


On the Actions page of a repository, under the “All workflows”, generally the following workflows are listed: 


The workflow files exist on any branches, and the workflows have been triggered …
  </blockquote>
</aside>

<p>In addition, if you want the workflow “<strong>GitHub Linux Release</strong>” is triggered after the workflow “<strong>CreateRelease</strong>” completed, you should only use the type “<strong>completed</strong>”.</p>
<p>If the problem still exists, please share your repository with us (if possible), so that we can check more detailed configurations related to the workflows to analyze the root cause.</p>

**Comment by sblantipodi** [comment]
<p>thanks for the answer, at the end I solved using a multi job workflow. but I understood your suggestion, thanks for the help. love GitHub and its community.</p>

**Reply by umassahu** [reply]
@sblantipodi : please let me know how did you solved this and what changes you have done w.r.t the example which you have given

**Comment by mundusnine** [comment]
<p>I encountered the issue on my repo <a href="https://github.com/Kode-Community/Kinc" rel="noopener nofollow ugc">here</a>.</p>
<p>So the workflows exist on the master but the default branch cant be the master because we depend on another functionality(wei/pull) to pull from the project that is forked.</p>
<p>Any suggestions as to why this would happen or how I could remedy it ?</p>

**Comment by Simran-B** [comment]
<p>The workflow dispatch UI on the website has a drop-down menu to select a branch. Maybe put some placeholder workflow on the default branch and the real workflow on the desired branch (here: master), with the exact name file name?</p>

**Comment by mundusnine** [comment]
<p>Ok so I am managing everything on the default branch now as actions  on master don’t work like they should.</p>
<p>Now the issue I am having is all the Windows actions get run but as soon as the last windows one is done(WindowsLibd3d12) the one that depends on that one(html5 library) doesn’t get called.</p>
<p>Could it be related to the fact that they use a different platform(one windows and the other ubuntu) or is it because libd3d12 is the last file and html5 library is the first file in the folder hierarchy ?</p>

**Comment by mundusnine** [comment]
<p>So after doing a Test it seems to be that workflow_run is capped at max 3 subsequent runs ? that is weird</p>

**Comment by Simran-B** [comment]
<p>It could be a deliberate limit to prevent endless loops of workflows being triggered, but if this is actually a restriction then it should be documented.</p>

**Comment by jamesamcl** [comment]
<p>Have also just hit this undocumented arbitrary limit of 3 consecutive runs using on workflow_run.</p>

**Comment by Yuqing-cat** [comment]
<p>Met The same problem and find its in the following document: <a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run" rel="noopener nofollow ugc">Events that trigger workflows - GitHub Docs</a>: <strong>Note:</strong> This event will only trigger a workflow run if the workflow file is on the default branch.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/3/a3161a967ba5cf6fd55084a1b06fdc7aafca6b63.png" data-download-href="/uploads/short-url/ngJbt4ozdzLaV2bL8FFm4OKOeJR.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/26561033/181090088-8060c998-b859-42cb-bdc5-42a9c7eef6c9.png" alt="image" data-base62-sha1="ngJbt4ozdzLaV2bL8FFm4OKOeJR" width="690" height="354" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3161a967ba5cf6fd55084a1b06fdc7aafca6b63_2_690x354.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3161a967ba5cf6fd55084a1b06fdc7aafca6b63_2_1035x531.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3161a967ba5cf6fd55084a1b06fdc7aafca6b63_2_1380x708.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/3/a3161a967ba5cf6fd55084a1b06fdc7aafca6b63_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">3005×1543 386 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Reply by jbcpollak** [reply]
does this mean the workflow will only _run_ on the default branch?

**Reply by swahili95** [reply]
yes

**Reply by FredFlitsPaal** [reply]
I finally figured it out for myself: it is not true the workflow only runs on de default branch. But it has to be on the default branch in order to execute the `workflow_run` trigger on any branch.

**Reply by sanjpras** [reply]
@FredFlitsPaal how were you able to do this? I have the workflow_run trigger on different branches, but the workflow only runs for the default branch.

**Reply by FredFlitsPaal** [reply]
I have the following on the default branch. Indenting is gone sorry about that.
`on:
  workflow_run:
    workflows: ["Run unit tests"]
    types:
      - completed`

Since this is present in the workflow file on the default branch, it works on other branches as well as long as the workflow filename is not changed.

**Comment by christopher-graham** [comment]
I am also seeing this issue, was there a resolution to it? 

My workflow files are all on the master (default) branch and I am not seeing my test workflow run after the initial build workflow completes.

**Comment by 95rade** [comment]
Is it possible to use workflow_run: trigger from a workflow in different repository? I think this is more of a 'real use case' scenario.

**Comment by balinteu** [comment]
Unfortunately almost 5 years after this post Github still has **not** changed the way workflow_run behaves. Workflow_run can be triggered from any branch, but the targeted workflow will **always** be executed on the **default** branch.

To fix this behaviour and actually do what I (and most of us) expect it to do, I added the followiong to the checkout action:

- name: Checkout Repository
      uses: actions/checkout@v4
      with:
        ref: ${{ github.event.workflow_run.head_branch }}

This checks out the branch from which the workflow_run call is being made. 

Yes I had to first add everything into the main branch, yes it is annoying and confusing, but it is what it is.
